module/VuFindSearch/src/VuFindSearch/Backend/Blender/Response/Json/RecordCollection.php

```diff
@@ -0,0 +1,441 @@
+                        $value,
+                        $facetType,
+                        $unmappedRule,
+                        $valueMap,
+                        $hierarchical
+                    );
+                    if ('' === $value) {
+                        continue;
+                    }
+
+                    if (isset($list[$value])) {
+                        $list[$value] += $count;
+                    } else {
+                        $list[$value] = intval($count);
+                    }
+
+                    if ($hierarchical) {
+                        $parts = explode('/', $value);
+                        $level = array_shift($parts);
+                        for ($i = $level - 1; $i >= 0; $i--) {
+                            $key = $i . '/'
+                                . implode('/', array_slice($parts, 0, $i + 1))
+                                . '/';
+                            if (isset($list[$key])) {
+                                $list[$key] += $count;
+                            } else {
+                                $list[$key] = intval($count);
+                            }
+                        }
+                    }
+                }
+            }
+
+            // Re-sort the list
+            uasort(
+                $list,
+                function ($a, $b) {
+                    return $b - $a;
+                }
+            );
+
+            $mergedFacets[$facetField] = $list;
+        }
+
+        $mergedFacets['blender_backend'] = $this->getBlenderFacetStats($collections);
+
+        // Convert the array back to Solr-style array with two elements
+        $facetFields = [];
+        foreach ($mergedFacets as $facet => $values) {
+            $list = [];
+            foreach ($values as $key => $value) {
+                $list[] = [$key, $value];
+            }
+            $facetFields[$facet] = $list;
+        }
+
+        return $facetFields;
+    }
+
+    /**
+     * Get facet counts for Blender backend facet
+     *
+     * @param array $collections Collections
+     *
+     * @return array
+     */
+    protected function getBlenderFacetStats(array $collections): array
+    {
+        $result = [];
+        foreach (array_keys($this->config->Backends->toArray()) as $backendId) {
+            $result[$backendId]
+                = isset($collections[$backendId])
+                ? $collections[$backendId]->getTotal() : 0;
+        }
+        return $result;
+    }
+
+    /**
+     * Convert facets into an associative array format for processing
+     *
+     * @param RecordCollectionInterface $collection Collection
+     *
+     * @return array
+     */
+    protected function convertFacets(RecordCollectionInterface $collection)
+    {
+        $facets = $collection->getFacets();
+        if ($facets instanceof Facets) {
+            return $facets->getFieldFacets();
+        }
+        if ($collection instanceof EDSRecordCollection) {
```

Comment:
Could we somehow move this logic into EDSRecordCollection so that the EDS record collection does the work and we don't have to have backend-specific logic in the blender? Something like a `getNormalizedFacets` method that the blender can rely on across all backends, and which could cause an exception to be thrown if unsupported?

Reply:
That's a good idea. Yet I think we should consider just making getFacets() of all RecordCollection classes adhere to the behavior documented in RecordCollectionInterface. If there's anything else, like pivot facets in Solr, they should be accessed with a separate method in RecordCollection. 

In this light I'd consider convertFacets a temporary measure to avoid issues with getFacets methods that don't adhere to the spec.

Reply:
If the existing EDS implementation is not conforming to the specification, then I would definitely agree that we should fix it rather than build a workaround. I hadn't realized/remembered that there was a standard being ignored here. If you don't have time to look into that right now, would you mind opening a JIRA ticket so I can address it when time permits?

Reply:
@demiankatz The Solr implementation doesn't conform to the spec either, and it doesn't even return an array [even though it says so](https://github.com/vufind-org/vufind/blob/dev/module/VuFindSearch/src/VuFindSearch/Backend/Solr/Response/Json/RecordCollection.php#L129). I think both should be fixed.

Reply:
I'll open a JIRA ticket for that.

Reply:
https://vufind.org/jira/browse/VUFIND-1540